crates/glaredb/src/lib.rs

```diff
@@ -312,6 +313,76 @@ impl From<Result<SendableRecordBatchStream, ExecError>> for RecordStream {
     }
 }
 
+/// RowMap represents a single record in an ordered map.
+type RowMap = indexmap::IndexMap<String, ScalarValue>;
+
+/// RowMapBatch is equivalent to a row-based view of a record batch.
+pub struct RowMapBatch(Vec<RowMap>);
```

Comment:
I'd add a quick comment this should be used sparingly/for tests.

Also, we could use an `Arc<str>` instead of `String` for the key if we want to keep the size down.

Reply:
arc str! 

My other theory was to just use the bson library, but the bson value enum has fewer types than the scalar value type, so avoiding more downcasting than necessary seems good